method/application/list.js

```diff
@@ -0,0 +1,55 @@
+const pull = require('pull-stream')
+
+module.exports = function GroupApplicationList (server) {
+  return function groupApplicationList (groupId, accepted, cb) {
+    const queryGroupId = [
+      {
+        $filter: {
+          value: {
+            content: {
+              type: 'group/application',
+              groupId,
+              tangles: {
+                application: {
+                  root: null
+                }
+              }
+            }
+          }
+        }
+      }
+    ]
+    const queryAll = [
+      {
+        $filter: {
+          value: {
+            content: {
+              type: 'group/application',
+              tangles: {
+                application: {
+                  root: null
+                }
+              }
+            }
+          }
+        }
+      }
+    ]
+
+    pull(
+      server.query.read({ query: groupId ? queryGroupId : queryAll }),
+      pull.map(i => i.key),
+      pull.asyncMap(server.tribes.application.get),
+      pull.filter(i => {
+        if (accepted !== null) {
+          if (accepted === true) {
+            return i.addMember
+          } else return !i.addMember
+        } else return i
```

Comment:
I would write this more linearly like this (I might not need to check `addMember.length` because it's `undefined | [feedId]` right ? I think elsewhere in this module I always return an empty Array)

```js
      pull.filter(i => {
        if (accepted === undefined)  return true
        if (accepted === true) return i.addMember && i.addMember.length
        if (accepted === false) return !i.addMember || i.addMember.length === 0

        throw new Error('tribes.application.list expected accepted to be (undefined | true | false)') // this should probably be checked way at the start
     })
```

Reply:
Much better :)